mypy/subtypes.py

```diff
@@ -616,6 +617,30 @@ def g(x: int) -> int: ...
 
         In this case, the first call will succeed and the second will fail: f is a
         valid stand-in for g but not vice-versa.
+
+    allow_potential_compatibility:
+        By default, this function returns True if and only if left is
+        definitely compatible with right.
+
+        If this flag is set to True, we relax the checks so they return True
+        if the left is *potentially* compatible to right. For example, the
+        following two functions are normally incompatible:
+
+            f(x: int, y: str = "...", *args: bool) -> str
+            g(*args: int) -> int
+
+        However, they would be *potentially* compatible under certain conditions --
+        for example, if the user runs "f_or_g(3)". So, if this flag is
```

Comment:
Again, this will be probably more clear (at least for me), if you say that partial overlap means that _some_ calls that succeed for right, will also succeed for the left callable.